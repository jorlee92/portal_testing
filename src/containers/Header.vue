<template lang="html">
  <header class="level">
    <nav class="level-left">
      <router-link to="/signup" v-if="!isAuthed" class="nav-link">Signup</router-link>
      <router-link to='login' v-if="!isAuthed" class="nav-link">Login</router-link>
      <a href="#" v-on:click="logout()" v-if="isAuthed" class="nav-link">Logout</a>
    </nav>
  </header>
</template>

<script>
import { mapGetters } from 'vuex'
export default {
  name: 'pp-header',
  methods: {
    logout () {
      this.$auth.signOut()
    },
  },
  computed: {
    ...mapGetters([
      'isAuthed',
    ]),
  },
}
</script>

<style lang="sass" scoped>
.nav-link
  margin: 0.5em
</style>
