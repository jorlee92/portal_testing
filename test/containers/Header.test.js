import { shallow } from 'vue-test-utils'
import Header from '@/containers/Header.vue'

describe('Header', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(Header)
  })
  it('should render the Header correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
