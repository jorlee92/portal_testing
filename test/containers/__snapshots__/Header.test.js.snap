// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render the Header correctly 1`] = `
<header
  class="level"
>
  <div
    class="level-left"
  >
    
    this be the header
  
  </div>
</header>
`;
